### ISSUE TITLE
RNN generate command returns the input text

### ISSUE THREAD

**Issue by loretoparisi** [AlexeyAB/darknet#286]
First of all thank you very much for your effort in the darknet project and this fork. The original project does not help a lot to solve git issues, and this is very frustrating since that's one of the reasons why these repos are on github, so thank again 💯 

I'm then reporting here my issue, about the RNN and model's parameter configuration, that is referenced from https://github.com/pjreddie/darknet/issues/210

I have trained a new text rnn with the command

`./darknet rnn train cfg/rnn.train.cfg -file /root/my_text_dataset.txt`

I have used the default `rnn.train.cfg` configuration, with `batch=64` and `steps=1000,1500`.

The generation command was like

`./darknet rnn generate cfg/rnn.cfg ./my_model.weights -srand 0 -len 1000 -seed love`

This returns part of the text in the dataset (the exact text, i.e. not a new generated text) starting from the seed.

I have tried the example weights provided `shakespeare.weights` and this does not happen:

```
# ./darknet rnn generate cfg/rnn.cfg ./shakespeare.weights -srand 0 -len 100 -seed castle
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
		connected                             256  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Loading weights from ./shakespeare.weights...Done!
castles, for he shall be
    The suit for him that comes from me.
```

and 

```
# ./darknet rnn generate cfg/rnn.cfg ./shakespeare.weights -srand 0 -len 100 -seed love 
loves me to a night,
    Which so we may have readined for your grave.
    Put all the wine enemies of m
```

etc.

So it seems to be a configuration problem in `cfg/rnn.train.cfg` or something else. My configuration for training (`rnn.train.cfg` looks like:

```
[net]
subdivisions=1
inputs=256
batch = 64
momentum=0.9
decay=0.001
max_batches = 2000
time_steps=576
learning_rate=0.1
policy=steps
steps=1000,1500
scales=.1,.1

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[connected]
output=256
activation=leaky

[softmax]

[cost]
type=sse
```

while my config for `generate` was

```
[net]
subdivisions=1
inputs=256
batch = 1
momentum=0.9
decay=0.001
max_batches = 2000
time_steps=1
learning_rate=0.1
policy=steps
steps=1000,1500
scales=.1,.1

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[connected]
output=256
activation=leaky

[softmax]

[cost]
type=sse
```

**[UPDATE]**

I have tried to train a new model weights for `shakespeare` text. Following the generated text from my trained model

```
# ./darknet rnn generate cfg/rnn.cfg /root/rnn_shakespeare_model.weights -srand 0 -len 1000 -seed BRUTUS
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
		connected                             256  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Loading weights from /root/rnn_shakespeare_model.weights...Done!
BRUTUS. Mark him out of a curse?
    The second of it be but young in a child, and of his benediction
    shall never have been a man transform your hands.
    It add so admired them.
  JULIA. Why should you be
    The bloody mind be gone to use?
    And then I would not throw upon your honour.
    To-morrow to my love with self-four kingdom,
    Whose eye-besides the means of his across sounds,
    And made them hot to die to her his face,
    The rest and the sword which else can be much.  
    I know you hear, my lord.
  DUKE. Where is he? Why, this is not a horse,
    Or else he died.
  ROSALIND. Dear lords, with their love shall die presently.
    There is no sea of fear.
  GLOUCESTER. The sound that's coming for a drum.
    Thou wast so dark'ned in your own prophecy,
    The gates of Helen's sake, and hast thou lov'd.
    The house is not so bad as well awhile.
    My virtue is ours of our worthy sweet;
    Seek him and never break out them in you.
    We are our chains, and cure the m
```

and here the generated text from the web site weights:

```
# ./darknet rnn generate cfg/rnn.cfg ./shakespeare.weights -srand 0 -len 1000 -seed BRUTUS
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
		connected                             256  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Loading weights from ./shakespeare.weights...Done!
BRUTUS. Now, but he poison with the world.
    So I near me, you know he is now tender off
    The fearful gates, and heart in hand a sweet  
     Perchance shuts a d the Prince of Wales,
    Witness which he bade esteem'd it.
    And then she would be as mine own ends,
    Whose designs which the mouth's outward sons.
    I am sorry for your blood, the shadow of my part,
    Whose grief shall be the battle to a fine
    Of the most infection of a spite
    Whereon thy self and fall the struant;
    And when they strike this morning to our love.
    Therefore I love thee let me with a power
    Of such a time with your displeasure that
    And he shall stay for no ready.
    My brother wears your queen in love by sea,
    Which is my friends and hearts now dear in their
                                                                  Exeunt




SCENE II.
The frontier was this forest

Enter CLEOPATRA, LOVEL, and LORDS

  CAMILLO. A cursed love to my will.
  Beat. I thank it out that I can se
```

This was my training task:

```
# ./darknet rnn train cfg/rnn.train.cfg -file ./t8.shakespeare.txt 
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
		connected                             256  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Learning Rate: 0.1, Momentum: 0.9, Decay: 0.001, Inputs: 256 36864 576
1: 0.993459, 0.993459 avg, 0.100000 rate, 7.002335 seconds, 0.006754 epochs
2: 0.990699, 0.993183 avg, 0.100000 rate, 6.196016 seconds, 0.013508 epochs
...
1999: 0.425721, 0.420526 avg, 0.001000 rate, 5.477683 seconds, 13.500998 epochs
2000: 0.418986, 0.420372 avg, 0.001000 rate, 5.326619 seconds, 13.507752 epochs
Saving weights to /root/rnn.backup
Saving weights to /root/rnn_final.weights
```

In both cases the (generated) output text seems not to belong to the input text:

```
# cat ./t8.shakespeare.txt | grep "ROSALIND. Dear lords"
# cat ./t8.shakespeare.txt | grep "CAMILLO. A cursed love"
```

while in my previous model I get the generated text in the input text like:

```
# cat /root/my_text_dataset.txt | grep "I would never claim to be no"
I would never claim to be no
I would never claim to be no, Ray Benzino
```

Since I'm using the same train configuration file `rnn.train.cfg` this could be due to the different datasets sizes:

```
# wc -l < /root/my_text_dataset.txt
200006
# wc -l < ./t8.shakespeare.txt 
124456
# ls -l ./t8.shakespeare.txt
-rw-r--r-- 1 root root 5458199 Mar 10  2009 ./t8.shakespeare.txt
# ls -l /root/my_text_dataset.txt
-rw-r--r-- 1 root root 7567372 Sep 25 08:41 /root/my_text_dataset.txt
```

**Comment by AlexeyAB** [comment]
@loretoparisi Hi,

* Do you use this fork or original repo?
* Can you provide your `my_model.weights`?
* What can you get using `./darknet rnn generate cfg/rnn.cfg ./my_model.weights -len 1000 -seed love` ?
* What can you get using `./darknet rnn generatetactic cfg/rnn.cfg ./my_model.weights -len 1000` ?

**Comment by loretoparisi** [comment]
@AlexeyAB thanks for your help I will try again and let you know. I was using the main repo, but now I'm trying your fork. Thanks for your effort for making darknet more useful!
